src/RulesTest/RuleRunnerTests.cs

```diff
@@ -343,6 +345,20 @@ public void RunAll_ReturnsExpectedResults()
             providerMock.VerifyAll();
         }
 
+        [TestMethod]
+        public void RunAll_ThrowsOnCancellation()
+        {
+            var e = new MockA11yElement();
+            var ruleMock = new Mock<IRule>(MockBehavior.Strict);
+            var providerMock = new Mock<IRuleProvider>(MockBehavior.Strict);
+            providerMock.Setup(m => m.All).Returns(() => new IRule[] { ruleMock.Object }).Verifiable();
+
+            var runner = new RuleRunner(providerMock.Object);
+            var cancellationToken = new CancellationTokenSource();
+            cancellationToken.Cancel();
+            var exception = Assert.ThrowsException<OperationCanceledException>(() => runner.RunAll(e, cancellationToken.Token));
```

Comment:
We can simplify this if we don't need to use the exception:
```suggestion
            Assert.ThrowsException<OperationCanceledException>(() => runner.RunAll(e, cancellationToken.Token));
```